web/src/app/administration/components/import-export/import-export.component.ts

```diff
@@ -40,68 +43,65 @@ export class ImportExportComponent implements OnInit {
     });
   }
 
-  export(domain = '') {
+  export(domain = ''): Observable<WorkbasketDefinition[] | Classification[]> {
     if (this.currentSelection === TaskanaType.WORKBASKETS) {
-      this.workbasketDefinitionService.exportWorkbaskets(domain);
+      return this.workbasketDefinitionService.exportWorkbaskets(domain);
     } else {
-      this.classificationDefinitionService.exportClassifications(domain);
+      return this.classificationDefinitionService.exportClassifications(domain);
     }
   }
 
   uploadFile() {
     const file = this.selectedFileInput.nativeElement.files[0];
-    const formdata = new FormData();
-    const ajax = new XMLHttpRequest();
+    console.log(this.selectedFileInput);
+    const formData = new FormData();
+    const xhr = new XMLHttpRequest();
     if (this.checkFormatFile(file)) {
-      formdata.append('file', file);
-      ajax.upload.addEventListener('progress', this.progressHandler.bind(this), false);
-      ajax.addEventListener('load', this.resetProgress.bind(this), false);
-      ajax.addEventListener('error', this.onFailedResponse.bind(this, ajax), false);
-      ajax.onreadystatechange = this.onReadyStateChangeHandler.bind(this, ajax);
+      formData.append('file', file);
+      xhr.upload.addEventListener('progress', this.progressHandler.bind(this), false);
+      xhr.addEventListener('load', this.resetProgress.bind(this), false);
+      xhr.addEventListener('error', this.onFailedResponse.bind(this, xhr), false);
+      xhr.onreadystatechange = this.onReadyStateChangeHandler.bind(this, xhr);
       if (this.currentSelection === TaskanaType.WORKBASKETS) {
-        ajax.open('POST', `${environment.taskanaRestUrl}/v1/workbasket-definitions`);
+        xhr.open('POST', `${environment.taskanaRestUrl}/v1/workbasket-definitions`);
       } else {
-        ajax.open('POST', `${environment.taskanaRestUrl}/v1/classification-definitions`);
+        xhr.open('POST', `${environment.taskanaRestUrl}/v1/classification-definitions`);
       }
       if (!environment.production) {
-        ajax.setRequestHeader('Authorization', 'Basic YWRtaW46YWRtaW4=');
+        xhr.setRequestHeader('Authorization', 'Basic YWRtaW46YWRtaW4=');
       }
-      ajax.send(formdata);
-      this.uploadservice.isInUse = true;
-      this.uploadservice.setCurrentProgressValue(1);
+      xhr.send(formData);
+      this.uploadService.isInUse = true;
+      this.uploadService.setCurrentProgressValue(1);
     }
   }
 
   progressHandler(event) {
     const percent = (event.loaded / event.total) * 100;
-    this.uploadservice.setCurrentProgressValue(Math.round(percent));
+    this.uploadService.setCurrentProgressValue(Math.round(percent));
   }
 
   private checkFormatFile(file): boolean {
-    const ending = file.name.match(/\.([^.]+)$/)[1];
```

Comment:
I don't know what this means but if this is the same as file.name.endsWith(), then your new version is great 😄